multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -185,41 +185,43 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	r.leaderNamespace = remoteCluster.GetNamespace()
 	r.leaderClusterID = string(remoteCluster.GetClusterID())
 
-	svc := &corev1.Service{}
-	if err := r.Client.Get(ctx, req.NamespacedName, &svcExport); err != nil {
-		klog.V(2).ErrorS(err, "Unable to fetch ServiceExport", "serviceexport", req.String())
-		if !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, err
-		}
+	cleanup := func() error {
 		if svcInstalled {
 			err = r.handleServiceDeleteEvent(ctx, req, remoteCluster)
 			if err != nil {
-				return ctrl.Result{}, err
+				return err
 			}
 			r.installedSvcs.Delete(svcObj)
 		}
-
 		if epInstalled {
 			err = r.handleEndpointDeleteEvent(ctx, req, remoteCluster)
 			if err != nil {
-				return ctrl.Result{}, err
+				return err
 			}
 			r.installedEps.Delete(epObj)
 		}
+		return nil
+	}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, &svcExport); err != nil {
+		klog.V(2).ErrorS(err, "Unable to fetch ServiceExport", "serviceexport", req.String())
+		if !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		if err := cleanup(); err != nil {
```

Comment:
Should we define cleanup() before this line?

Reply:
A following line also use this method. https://github.com/antrea-io/antrea/pull/3652/files#diff-1e1d6e2558b5bb86a952c580e877a8a072c499a0c278194afb4d0948ae459d10R223

Reply:
Ok.